wayland-cursor/src/lib.rs

```diff
@@ -216,14 +198,161 @@ impl<'a> Cursor<'a> {
 ///
 /// Note that this proxy will be considered as "unmanaged" by the crate, as such you should
 /// not try to act on it beyond assigning it to `wl_surface`s.
-pub struct CursorImageBuffer<'a> {
-    _cursor: PhantomData<&'a Cursor<'a>>,
+#[derive(Clone)]
+pub struct CursorImageBuffer {
     buffer: WlBuffer,
+    delay: u32,
+    xhot: u32,
+    yhot: u32,
+    width: u32,
+    height: u32,
 }
 
-impl<'a> Deref for CursorImageBuffer<'a> {
+impl CursorImageBuffer {
+    /// Construct a new CursorImageBuffer
+    ///
+    /// This function appends the pixels of the image to the provided file,
+    /// and constructs a wl_buffer on that data.
+    fn new(theme: &mut CursorTheme, image: &xcur::parser::Image) -> Self {
+        let buf = CursorImageBuffer::convert_pixels(&image.pixels);
+        let offset = theme.file.seek(SeekFrom::End(0)).unwrap();
+        theme.file.write_all(&buf).unwrap();
+
+        let new_size = theme.file.seek(SeekFrom::End(0)).unwrap();
+        theme.grow(new_size as i32);
+
+        let buffer = theme
+            .pool
+            .create_buffer(
+                offset as i32,
+                image.width as i32,
+                image.height as i32,
+                (image.width * 4) as i32,
+                Format::Argb8888,
+            )
+            .detach();
+
+        CursorImageBuffer {
+            buffer,
+            delay: image.delay,
+            xhot: image.xhot,
+            yhot: image.yhot,
+            width: image.width,
+            height: image.height,
+        }
+    }
+
+    /// Convert the pixels saved in `u32`s into `u8`s.
+    fn convert_pixels(pixels: &[u32]) -> Vec<u8> {
+        let mut buf: Vec<u8> = Vec::with_capacity(pixels.len() * 4);
+
+        for pixel in pixels {
+            buf.push((pixel >> 24) as u8);
+            buf.push((pixel >> 16) as u8);
+            buf.push((pixel >> 8) as u8);
+            buf.push(*pixel as u8);
+        }
+
+        buf
+    }
+}
```

Comment:
We can use [`byteorder`](https:://docs.rs/byteorder) to handle that. `wayland-client` already depends on it anyway.

Reply:
> We can use [`byteorder`](https:://docs.rs/byteorder) to handle that. `wayland-client` already depends on it anyway.

How did github go from https:://docs.rs/byteorder to
https://github.com/Smithay/wayland-rs/pull/://docs.rs/byteorder to
https://github.com/Smithay/wayland-rs/compare/:/docs.rs/byteorder?expand=1
even now it continues to surprise me every day